catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/EntityResourceTest.java

```diff
@@ -470,6 +482,96 @@ protected void prepareGetWithDifferentFields(T entity) throws HttpResponseExcept
   ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
   // Common entity tests for GET operations
   ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+  @Test
+  void get_entityWithAuthorizedViewMetadataOperation(TestInfo test)
+      throws HttpResponseException, JsonProcessingException {
+    if (!this.getClass().getName().endsWith("TableResourceTest")) {
+      return;
+    }
+
+    // Create an entity for testing access to view metadata.
+    T entity = createEntity(createRequest(test, 0), ADMIN_AUTH_HEADERS);
+    EntityInterface<T> entityInterface = getEntityInterface(entity);
+    String originalEntity = JsonUtils.pojoToJson(entity);
+    entityInterface.setOwner(USER_OWNER1);
+    patchEntity(entityInterface.getId(), originalEntity, entityInterface.getEntity(), ADMIN_AUTH_HEADERS);
+
+    // Disable DataConsumer ViewMetadata permission.
```

Comment:
Note to self: This needs to be modified. 

DataConsumer should have ViewMetadata permission as is
DataConsumer should not have ViewMetadata on certain set of tables
Finance should have ViewMetadata to a certain set of tables

Reply:
Will need to update permissions to and condition of result of policy evaluation, so that if any of the permissions yields allow:false (DENY), then return DENY.

Reply:
There are more complexities here on how this can be modeled. 

The above scenario does not work.

Nor does it work with jcasbin

![Uploading image.png…]()

Reply:
Using priorities would yield intended results

<img width="1269" alt="image" src="https://user-images.githubusercontent.com/5906984/156904566-be9ae09c-411e-4ed8-be50-31e31c6dc090.png">

Reply:
After a few tries I'm more convinced that priority is the only way to program this to yield a reliable outcome.

Reply:
Good resource here:
https://github.com/casbin/casbin/issues/550

Reply:
One of the option we used in streamline is to build a ACL table that will take the object_id( in this case entity id) and put a mapping with either user_id, role or team_id.
```
CREATE TABLE IF NOT EXISTS acl_entry (
  id              BIGINT AUTO_INCREMENT NOT NULL,
  objectId        BIGINT                NOT NULL,
  objectNamespace VARCHAR(255)          NOT NULL,
  sidId           BIGINT                NOT NULL,
  sidType         VARCHAR(255)          NOT NULL,
  permissions     TEXT                  NOT NULL,
  owner           BOOLEAN               NOT NULL,
  `grant`         BOOLEAN               NOT NULL,
  timestamp       BIGINT,
  PRIMARY KEY (id)
);
```
objectId -> entityId
objectNamespace -> entityType ( might not be necessary)
sidId -> user_id,team_id,role
permissions -> read, write, edit
owner -> is the sidId is owner of this entity

Reply:
The above will give maximum flexibility to enforce the permissions at finer grain. i.e one can provide permissions at column level and each permission becomes a row per sidID